### DISCUSSION TITLE
Auto suggestion list is not showing

### DISCUSSION THREAD

**Discussion by Arsik0153** [zed-industries/zed#19516] [Feature Requests]
Hi. I stumbled upon this problem:
I'm trying to open autocomplete menu that should open on Ctrl + Space. When hitting this shortcut, nothing happens. I made sure this shortcut is not in use by Mac. I also tried to bind to another key: 
```
"bindings": {
    "cmd-k": "editor::ShowCompletions"
  }
```
  It didn't work for me. Looking for help

**Comment by mocenigo** [comment]
It depends a lot on 1) whether you have something that actually can show the completions (such as a language server and the corresponding client) and 2) if it is installed whether it is still doing something else.

Regarding 2), for instance, you see completions automatically if you have ltex with latex or markdown, but only after it has finished parsing the other documents. Each time you save, it parses everything again, and completions do not show. Apparently, incremental completion database polling and multithreading are not implemented for most language servers, and/or the zed implementation lacks a bit in this regard.

**Reply by Arsik0153** [reply]
@mocenigo I am pretty sure language server is working, because if I wait long enough (~5 sec), then autocompletion list is showing.  However, I was expecting different behavior, like in VSCode, where if we click Ctrl + Space, autocompletion list is showing. Maybe it's the behavior that was intended? If yes, why there's `editor::ShowCompletions` keybinding?

**Reply by mocenigo** [reply]
> @mocenigo I am pretty sure language server is working, because if I wait long enough (~5 sec), then autocompletion list is showing. However, I was expecting different behavior, like in VSCode, where if we click Ctrl + Space, autocompletion list is showing. Maybe it's the behavior that was intended? If yes, why there's `editor::ShowCompletions` keybinding?

Then it is the second case. Completion support on zed suffers from the issue I have decsribed. Maybe @notpeter, @mrnugget ot @mikayla-maki can help?

~~In any case, @Arsik0153 , I would suggest you raise an issue here. This is a useability matter, and if there is something VSCode does smarter, Zed should do it as well.~~

Edit:  @Arsik0153 , this is https://github.com/zed-industries/zed/issues/12750 go there and click on tyhe thumb up.

**Reply by Arsik0153** [reply]
Seems there's no progress since late June. Anyways, thank you for help. Hope one day issue will be closed

**Reply by mocenigo** [reply]
> Seems there's no progress since late June. Anyways, thank you for help. Hope one day issue will be closed

The developers are very busy. They need a bit of help from our side, and I am planning to give.

**Reply by Arsik0153** [reply]
If I can help you fix this problem or you need information about my mac, let me know. Not familiar with rust btw

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
